python/src/ccf/ledger.py

```diff
@@ -156,88 +151,90 @@ class PublicDomain:
     All public tables within a :py:class:`ccf.ledger.Transaction`.
     """
 
-    _buffer: io.BytesIO
-    _buffer_size: int
+    _buffer: bytes
+    _cursor: int
     _entry_type: EntryType
     _claims_digest: bytes
     _version: int
     _max_conflict_version: int
     _tables: dict
 
-    def __init__(self, buffer: io.BytesIO):
+    def __init__(self, buffer: bytes):
+        self._entry_type = EntryType(buffer[0])
+
+        self._cursor = 1
```

Comment:
why 1?

Reply:
Because we've read 1-byte already for the `entry_type`. Doing a 1-byte struct unpack on this was suprisingly slow, so we just use the raw byte. Added a comment.